src/Blueprint/Blueprint/Cmdlets/SetAzureRMBlueprintAssignment.cs

```diff
@@ -0,0 +1,124 @@
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public Hashtable ResourceGroup { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.Parameters)]
+        [ValidateNotNull]
+        public Hashtable Parameter { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, HelpMessage = ParameterHelpMessages.SystemAssignedIdentity)]
+        public SwitchParameter SystemAssignedIdentity { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.UserAssignedIdentity)]
+        [ValidateNotNullOrEmpty]
+        public string UserAssignedIdentity { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, HelpMessage = ParameterHelpMessages.LockFlag)]
+        public PSLockMode? Lock { get; set; }
+        #endregion Parameters
+
+        #region Cmdlet Overrides
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                var subscriptionsList = SubscriptionId ?? new[] { DefaultContext.Subscription.Id };
+
+                if (ShouldProcess(string.Join(",", subscriptionsList), string.Format(Resources.UpdateAssignmentShouldProcessString, Name)))
+                {
+                    // If explicitly requested to use user assigned identity let's do that, otherwise let's default to system assigned
+                    if (this.IsParameterBound(c => c.UserAssignedIdentity))
+                    {
+                        var userAssignedIdentity = new Dictionary<string, UserAssignedIdentity>()
+                        {
+                            { UserAssignedIdentity, new UserAssignedIdentity() }
+                        };
+
+                        var assignment = CreateAssignmentObject(ManagedServiceIdentityType.UserAssigned,
+                            userAssignedIdentity,
+                            Location,
+                            Blueprint.Id,
+                            Lock,
+                            Parameter,
+                            ResourceGroup);
+
+                        foreach (var subscription in subscriptionsList)
+                        {
+                            var scope = Utils.GetScopeForSubscription(subscription);
+                            ThrowIfAssignmentNotExist(scope, Name);
+                            // Register Blueprint RP
+                            RegisterBlueprintRp(subscription);
+
+                            WriteObject(BlueprintClient.CreateOrUpdateBlueprintAssignment(scope, Name, assignment));
```

Comment:
@filizt is the underlying  call for `CreateOrUpdateBlueprintAssignment` a `PUT` or a `PATCH`? If it's a `PATCH`, then we should rename the cmdlet to be `Update-*` instead of `Set-*`

Reply:
It's a PUT call

---
In reply to: [261403792](https://github.com/Azure/azure-powershell/pull/8537#discussion_r261403792) [](ancestors = 261403792)